### PR TITLE
libalsa: Update SPDX license identifier

### PR DESCRIPTION
**Pull request by jwillikers**
Use rm_safe functions in the configure method.

Specify library name and version:  **libalsa/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libalsa/all/conanfile.py [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
 import os
 
-required_conan_version = ">=1.52.0"
+required_conan_version = ">=1.53.0"
 
 
 class LibalsaConan(ConanFile):
     name = "libalsa"
-    license = "LGPL-2.1"
+    license = "LGPL-2.1-only"
```
[the remark this answers is not in the document]
Comment (jwillikers) [reply recipes/libalsa/all/conanfile.py:15]: @prince-chrismc That text just describes that a library _may_ say it allows `"any later version"`. I couldn't find anything that said the license might apply to `"any later version"` in the README or documentation.

[the remark this answers is not in the document]
Comment (jwillikers) [reply recipes/libalsa/all/conanfile.py:15]: Wait, I found it mentioned in a header file: https://github.com/alsa-project/alsa-lib/blob/master/src/async.c#L12-L14

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/libalsa/all/conanfile.py]: The LGPL stuff is a mess --- the only think that makes then differ is letter they say "any later" in the license text itself

https://spdx.org/licenses/LGPL-2.1-or-later.html

>13. The Free Software Foundation may publish revised and/or new versions of the Lesser General Public License from time to time. Such new versions will be similar in spirit to the present version, but may differ in detail to address new problems or concerns.
Each version is given a distinguishing version number. If the Library specifies a version number of this License which applies to it and "any later version", you have the option of following the terms and conditions either of that version or of any later version published by the Free Software Foundation. If the Library does not specify a license version number, you may choose any version ever published by the Free Software Foundation.

https://spdx.org/licenses/LGPL-2.1-only

😭  they have the same text

> 13. The Free Software Foundation may publish revised and/or new versions of the Lesser General Public License from time to time. Such new versions will be similar in spirit to the present version, but may differ in detail to address new problems or concerns.
Each version is given a distinguishing version number. If the Library specifies a version number of this License which applies to it and "any later version", you have the option of following the terms and conditions either of that version or of any later version published by the Free Software Foundation. If the Library does not specify a license version number, you may choose any version ever published by the Free Software Foundation.

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/libalsa/all/conanfile.py]: Yep you are right -- that one is distinguished by the actual header that goes in the code! 🤢 

Alrighty the more you know

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**